app/HutchManager/Services/AgentService.cs

```diff
@@ -110,4 +113,33 @@ public async Task Delete(int agentId)
     _db.Agents.Remove(entity);
     await _db.SaveChangesAsync();
   }
+
+  /// <summary>
+  /// Generate a new client id and secret 
+  /// </summary>
+  /// <param name="isNew"></param>
+  /// <returns></returns>
+  /// <exception cref="KeyNotFoundException"></exception>
+  public async Task<ManageAgent> Generate (bool isNew)
+  {
+    // Generate a unique Client Id
+    var clientId = string.Empty;
+    var isClientIdUnique = false;
+    while (!isClientIdUnique)
+    {
+      var uniqueClientId = Crypto.GenerateId();
+      var existingClientIds = await _db.Agents.AnyAsync(x => x.ClientId == uniqueClientId);
+      if (existingClientIds) continue;
+      clientId = uniqueClientId;
+      isClientIdUnique = true;
+    }
```

Comment:
Is there a risk here that the `Cryto.GenerateId()` will generate a non-unique ID? Did you encounter this when testing?